release-notes/6.0/preview/api-diff/preview1/.Net/6.0-preview1_System.IO.md

```diff
@@ -0,0 +1,49 @@
+# System.IO
+
+``` diff
+ namespace System.IO {
+     public class BinaryReader : IDisposable {
++        public virtual Half ReadHalf();
+     }
+     public class BinaryWriter : IAsyncDisposable, IDisposable {
++        public virtual void Write(Half value);
+     }
+     public sealed class BufferedStream : Stream {
+-        public override int Read(byte[] array, int offset, int count);
++        public override int Read(byte[] buffer, int offset, int count);
+-        public override void Write(byte[] array, int offset, int count);
++        public override void Write(byte[] buffer, int offset, int count);
+     }
+     public class FileStream : Stream {
+-        public override IAsyncResult BeginRead(byte[] array, int offset, int numBytes, AsyncCallback callback, object state);
++        public override IAsyncResult BeginRead(byte[] buffer, int offset, int count, AsyncCallback callback, object state);
+-        public override IAsyncResult BeginWrite(byte[] array, int offset, int numBytes, AsyncCallback callback, object state);
++        public override IAsyncResult BeginWrite(byte[] buffer, int offset, int count, AsyncCallback callback, object state);
+-        public override int Read(byte[] array, int offset, int count);
++        public override int Read(byte[] buffer, int offset, int count);
+-        public override void Write(byte[] array, int offset, int count);
```

Comment:
The parameter names here differ from the base method they're overriding, so if you wanted to specify a parameter name at the call site, you needed to use a different parameter name based on the type of the variable through which you were making a call.  On top of that, if you passed in an invalid argument, you would frequently get an exception giving you a different name than what was actually in the metadata for the derived type, because the exception would be coming from the base method.  And having different names meant everyone ended up needing to write and rewrite their own validation logic, whereas when everyone appropriately used the same name, it could all be put into a helper method that everyone then uses and which is now exposed publicly (and we could write a stream validation test suite that asserted those names for all streams).